### PR TITLE
python310Packages.sipyco: init at version 1.4

### PR DESCRIPTION
**Pull request by charlesbaynham**
###### Description of changes

Add sipyco (https://github.com/m-labs/sipyco), a package for working with scientific hardware, used heavily by ARTIQ (https://github.com/m-labs/artiq). 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/sipyco/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  ];
+
+  pythonImportsCheck = [
+    "sipyco"
+  ];
+
+  meta = with lib; {
+    description = "Simple Python Communications - used by the ARTIQ experimental control package";
+    homepage = "https://github.com/m-labs/sipyco";
+    changelog = "https://github.com/m-labs/sipyco/releases/tag/${version}";
```
Comment (natsukium) [inline pkgs/development/python-modules/sipyco/default.nix:37]:
```suggestion
    changelog = "https://github.com/m-labs/sipyco/releases/tag/v${version}";
```

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 227876` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.sipyco</li>
    <li>python310Packages.sipyco.dist</li>
    <li>python311Packages.sipyco</li>
    <li>python311Packages.sipyco.dist</li>
  </ul>
</details>

**Comment by natsukium** [comment]
`python3Packages.sipyco: init at 1.4` is preferred for the commit message.

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.sipyco

**Pull request merged**